src/lib.rs

```diff
@@ -753,17 +753,21 @@ impl<T: Counter> Histogram<T> {
     ///
     /// Returns an error if `value` exceeds the highest trackable value and auto-resize is
     /// disabled.
-    pub fn record_n_correct(&mut self, value: i64, count: T, interval: i64) -> Result<(), ()> {
+    pub fn record_n_correct(&mut self, value: u64, count: T, interval: u64) -> Result<(), ()> {
         try!(self.record_n(value, count));
-        if interval <= 0 {
+        if interval == 0 {
             return Ok(());
         }
 
-        let mut missingValue = value - interval;
-        while missingValue >= interval {
-            try!(self.record_n(missingValue, count));
-            missingValue -= interval;
+        if value > interval {
+            // only enter loop when calculations will stay non-negative
+            let mut missingValue = value - interval;
+            while missingValue >= interval {
+                try!(self.record_n(missingValue, count));
+                missingValue -= interval;
```

Comment:
Regarding your (now "outdated") comment on this loop -- I think not doing the last interval (in other words, the current logic) is correct, given the intent of this method.

Suppose the expected interval is 5, and the value we have is 17. The current logic would record 17, 12, 7, and stop: the lowest recorded value would be in `[expected interval, 2 * expected - 1]`. 

The alternate logic that includes an extra value would result in also recording the value 2. I think this is not right, because that's basically going to skew your latency numbers to be *better* than expected. It's certainly possible that in some cases this would be desirable, but I'd wager that in general users would be surprised by this (and not want it).